lib/browser/guest-window-manager.js

```diff
@@ -81,6 +85,40 @@ const mergeBrowserWindowOptions = function (embedder, options) {
   return options
 }
 
+const MULTIPART_CONTENT_TYPE = 'multipart/form-data'
+const URL_ENCODED_CONTENT_TYPE = 'application/x-www-form-urlencoded'
+function makeContentTypeHeader ({ contentType, boundary }) {
+  const header = `content-type: ${contentType};`
+  if (contentType === MULTIPART_CONTENT_TYPE) {
+    return `${header} boundary=${boundary}`
+  }
+  return header
+}
+
+// Figure out appropriate headers for post data.
+const parseContentTypeFormat = function (postData) {
+  if (postData.length) {
+    // For multipart forms, the first element will start with the boundary
+    // notice, which looks something like `------WebKitFormBoundary12345678`
+    // Note, this regex would fail when submitting a urlencoded form with an
+    // input attribute of name="--theKey", but, uhh, don't do that?
+    const postDataFront = postData[0].bytes.toString()
```

Comment:
Why is it OK to assume that `postData[0].bytes` is defined here (as opposed to `filePath`, `blobUUID`, or any of the other options for specifying data on a PostData object)?

Reply:
Because the first element in the postData array (if it's an array, it means it's a multipart form) is always describing the boundary, not the actual content (which would be the next element in the array).